.github/workflows/CompatHelper.yml

```diff
@@ -1,24 +1,44 @@
 name: CompatHelper
-
 on:
   schedule:
-    - cron: '00 00 * * *'
-
+    - cron: 0 0 * * *
+  workflow_dispatch:
+permissions:
+  contents: write
+  pull-requests: write
 jobs:
   CompatHelper:
-    runs-on: ${{ matrix.os }}
-    strategy:
-      matrix:
-        julia-version: [1.2.0]
-        julia-arch: [x86]
-        os: [ubuntu-latest]
+    runs-on: ubuntu-latest
     steps:
-      - uses: julia-actions/setup-julia@latest
+      - name: Check if Julia is already available in the PATH
+        id: julia_in_path
+        run: which julia
+        continue-on-error: true
+      - name: Install Julia, but only if it is not already available in the PATH
+        uses: julia-actions/setup-julia@v1
         with:
-          version: ${{ matrix.julia-version }}
-      - name: Pkg.add("CompatHelper")
-        run: julia -e 'using Pkg; Pkg.add("CompatHelper")'
-      - name: CompatHelper.main()
+          version: '1'
+          arch: ${{ runner.arch }}
+        if: steps.julia_in_path.outcome != 'success'
+      - name: "Add the General registry via Git"
+        run: |
+          import Pkg
+          ENV["JULIA_PKG_SERVER"] = ""
+          Pkg.Registry.add("General")
+        shell: julia --color=yes {0}
+      - name: "Install CompatHelper"
+        run: |
+          import Pkg
+          name = "CompatHelper"
+          uuid = "aa819f21-2bde-4658-8897-bab36330d9b7"
+          version = "3"
+          Pkg.add(; name, uuid, version)
+        shell: julia --color=yes {0}
+      - name: "Run CompatHelper"
+        run: |
+          import CompatHelper
+          CompatHelper.main()
+        shell: julia --color=yes {0}
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        run: julia -e 'using CompatHelper; CompatHelper.main()'
+          COMPATHELPER_PRIV: ${{ secrets.DOCUMENTER_KEY }}
```

Comment:
@DilumAluthge - I copied the code verbatim as it was given in the instructions. However I am not sure if line:
```
 COMPATHELPER_PRIV: ${{ secrets.DOCUMENTER_KEY }}
```
is indeed correct? Why do we use `DOCUMENTER_KEY` here?

Also - do we need to set any environment variables in DataFrames.jl (and if yes - how)?

Thank you!

Reply:
I should probably put a comment in the YAML file to explain this.

You just need to make sure that you are providing the SSH private key from an SSH deploy keypair that has write access to this repository.

Many people already have such a private key, which they use for Documenter.jl. So that’s the default name I use in the template. But you should change the name to match whatever you have named the GitHub Secret in the repository settings (https://github.com/JuliaData/DataFrames.jl/settings).

Reply:
Excellent - thank you! We have Documenter 😄.